app/src/main/java/com/nextcloud/talk/utils/PushUtils.java

```diff
@@ -226,162 +221,159 @@ public void pushRegistrationToServer() {
         String token = appPreferences.getPushToken();
 
         if (!TextUtils.isEmpty(token)) {
-            String credentials;
             String pushTokenHash = generateSHA512Hash(token).toLowerCase();
             PublicKey devicePublicKey = (PublicKey) readKeyFromFile(true);
             if (devicePublicKey != null) {
-                byte[] publicKeyBytes = Base64.encode(devicePublicKey.getEncoded(), Base64.NO_WRAP);
-                String publicKey = new String(publicKeyBytes);
-                publicKey = publicKey.replaceAll("(.{64})", "$1\n");
+                byte[] devicePublicKeyBytes = Base64.encode(devicePublicKey.getEncoded(), Base64.NO_WRAP);
+                String devicePublicKeyBase64 = new String(devicePublicKeyBytes);
+                devicePublicKeyBase64 = devicePublicKeyBase64.replaceAll("(.{64})", "$1\n");
 
-                publicKey = "-----BEGIN PUBLIC KEY-----\n" + publicKey + "\n-----END PUBLIC KEY-----\n";
+                devicePublicKeyBase64 = "-----BEGIN PUBLIC KEY-----\n" + devicePublicKeyBase64 + "\n-----END PUBLIC KEY-----\n";
 
                 if (userUtils.anyUserExists()) {
-                    String providerValue;
-                    PushConfigurationState accountPushData = null;
                     for (Object userEntityObject : userUtils.getUsers()) {
                         UserEntity userEntity = (UserEntity) userEntityObject;
-                        providerValue = userEntity.getPushConfigurationState();
-                        if (!TextUtils.isEmpty(providerValue)) {
-                            try {
-                                accountPushData = LoganSquare.parse(providerValue, PushConfigurationState.class);
-                            } catch (IOException e) {
-                                Log.d(TAG, "Failed to parse account push data");
-                                accountPushData = null;
-                            }
-                        } else {
-                            accountPushData = null;
-                        }
 
-                        if (((TextUtils.isEmpty(providerValue) || accountPushData == null) && !userEntity.getScheduledForDeletion()) ||
-                                (accountPushData != null && !accountPushData.getPushToken().equals(token) && !userEntity.getScheduledForDeletion())) {
-
```

Comment:
i didn't like that if check, so first i refactored and came up with

![grafik](https://user-images.githubusercontent.com/2932790/150128714-82ec4cf8-11a6-4171-9bc4-0c648a415416.png)

After that, i reduced it to only check if user is marked for deletion. 
Because it doesn't hurt to update the token even if the device is still of the opinion that it should still be the same..?
what do you think @nickvergessen @tobiasKaminsky  ?

Reply:
@nickvergessen ?

Reply:
LGTM